scalafix-rules/src/main/scala/scalafix/internal/rule/ProcedureSyntax.scala

```diff
@@ -11,7 +11,8 @@ import scalafix.v1._
 class ProcedureSyntax extends SyntacticRule("ProcedureSyntax") {
 
   override def description: String =
-    "Replaces deprecated procedure syntax with explicit ': Unit ='"
+    "Replaces deprecated procedure syntax with explicit ': Unit ='" +
+      "This rule is specific to scala 2, since procedure syntax is not supported in Scala 3"
```

Comment:
I guess it goes back to the question of whether unsupported rules should be silently ignored or trigger a failure, but I would have expected this to come with a check like https://github.com/scalacenter/scalafix/blob/933b9403e269cad1133d65696821a5d387de9c86/scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala#L49-L93. In that case it's not a big deal since code not respecting the procedure syntax wouldn't compile anyway, but there might be other cases where we prefer a verbose failure instead of false negatives.

Reply:
Yes in fact in this case, there is no way to have procedure syntax in scala 3 code, so this rule wouldn't rewrite scala 3 code projects. 
But should we do that in other cases: for example in DisableSyntax, some options don't make sense anymore since  some syntax is not allowed anymore in scala 3. Should we warn about some options to say, that they won't have any effects ? 

- In this case, I agree we can throw an error this rule is not useful in Scala 3

Reply:
Maybe now is a good time to design/implement https://github.com/scalacenter/scalafix/issues/1199? I feel like it could address the recurring questions we have with built-in rules (as discussed in https://github.com/scalacenter/scalafix/pull/1394), and that community rules will probably have.